cmd/admin-handlers.go

```diff
@@ -1646,6 +1646,32 @@ func assignPoolNumbers(servers []madmin.ServerProperties) {
 	}
 }
 
+// GetObjectDebugInfoHandler - GET /minio/admin/v3/object
+// Get the entire set of metadata for an object.
+func (a adminAPIHandlers) GetObjectDebugInfoHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "GetObjectDebugInfoHandler")
+	defer logger.AuditLog(ctx, w, r, mustGetClaimsFromToken(r))
+	objectAPI, _ := validateAdminReq(ctx, w, r, iampolicy.StorageInfoAdminAction)
+	if objectAPI == nil {
+		return
+	}
+	bucket := r.URL.Query().Get("bucket")
+	object := r.URL.Query().Get("object")
+	debugInfo, err := objectAPI.GetObjectDebugInfo(ctx, bucket, object, ObjectOptions{})
+	if err != nil {
+		writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(toAPIErrorCode(ctx, err)), r.URL)
+		return
+	}
+	enc := json.NewEncoder(w)
+	err = enc.Encode(debugInfo)
+	if err != nil {
+		writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(toAPIErrorCode(ctx, err)), r.URL)
+		return
+	}
+	w.(http.Flusher).Flush()
+	return
+}
```

Comment:
This API should be called HealInspect() and should take `object` as a prefix also to do a recursive listing to return information back to the client like `AdminTrace` style. 

@kerneltime I spoke to @abperiasamy about this we should change this to a single call for all objects if asked for a path or a single object. 

This way an entire namespace dump can be obtained.